### DISCUSSION TITLE
Clarifying requirements for training from local hierarchy

### DISCUSSION THREAD

**Discussion by DrRajJena** [microsoft/InnerEye-DeepLearning#758] [General]
Folks. I followed the instructions for creating a local hierarchy to create new training runs. I copied what I thought was the minimum number of files needed.

My tree listing for subfolder InnerEyeFoo in InnerEyeDeepLearning
├── ML
│   ├── configs
│   │   └── segmentation
│   │       ├── HeadAndNeckBase.py
│   │       ├── ProstateAB.py
│   │       └── ProstateBase.py
│   └── runner.py
└── settings.yml

My settings.yml looks like this
subscription_id: <Subscription removed from here>
  application_id: ''
  azureml_datastore: <removed>
  resource_group: <removed>
  docker_shm_size: '440g'
  workspace_name: <removed>
  cluster: 'NC24'
  model_configs_namespace: 'InnerEyeFoo.ML.configs'
  extra_code_directory: 'InnerEyeFoo'

Running in the InnerEye conda environment, my runner.py cannot see InnerEye libraries, even though I can manually import them from the command line from the same path.

Could you advise if I have missed steps to create my local hierarchy?

**Comment by ant0nsc** [comment]
Hi @DrRajJena, I can't say for certain how to resolve that without knowing where your `InnerEye` folder lives. Is your setup like
```
InnerEye-DeepLearning
   > InnerEye
   > Tests
   > InnerEyeFoo
         > ML
             runner.py
```
If so, you would need to modify `sys.path` inside of `runner.py` _before_ you make your first import statement from the InnerEye codebase. It would look something like:
```
import sys
from pathlib import Path
# Here, you need to adjust how often you call .parent to get to the root of the project where the InnerEye folder is located.
root = Path(__file__).resolve().parent.parent

if (root / "InnerEye").is_dir():
    root_str = str(root)
    if root_str not in sys.path:
        print(f"Adding to sys.path: {root_str}")
        sys.path.insert(0, root_str)
else:
    raise ValueError("I'm expecting to find a folder called InnerEye in the parent folder of this file.")
```

**Reply by DrRajJena** [reply]
Perfect - thank you!
